File: app/src/main/java/com/ikvych/cocktail/data/repository/base/DrinkApiRepository.java
package com.ikvych.cocktail.data.repository.base;

import androidx.lifecycle.MutableLiveData;

import com.ikvych.cocktail.data.entity.Drink;

import java.util.List;

public interface DrinkApiRepository {

    void updateDrinksLiveData(String name);

    MutableLiveData<List<Drink>> getDrinksLiveData();

    List<Drink> getCurrentData();
}

File: app/src/main/java/com/ikvych/cocktail/data/repository/base/DrinkApiRepository.kt
package com.ikvych.cocktail.data.repository.base

import androidx.lifecycle.MutableLiveData
import com.ikvych.cocktail.data.entity.Drink


interface DrinkApiRepository {

    fun getLiveData(): MutableLiveData<List<Drink>>
    fun updateDrinksLiveData(query: String)
    fun getCurrentData(): List<Drink>
    fun initAllIngredient()
}